modules/system-disaster-recovery/src/main/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryStorage.java

```diff
@@ -97,4 +101,15 @@ void saveResetClusterMessage(ResetClusterMessage message) {
     public @Nullable ResetClusterMessage readVolatileResetClusterMessage() {
         return volatileResetClusterMessage;
     }
+
+    @Override
+    public @Nullable UUID readWitnessedMetastorageRepairClusterId() {
+        VaultEntry entry = vault.get(WITNESSED_METASTORAGE_REPAIR_CLUSTER_ID_VAULT_KEY);
```

Comment:
Just in case. We do replicate id/index with cmg raft snapshot, right?

Reply:
We do, take a look at `CmgRaftGroupListener#onSnapshotSave()`